dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java

```diff
@@ -529,6 +529,39 @@ public void removeTaskLogFile(Integer processInstanceId) {
         }
     }
 
+    /**
+     * recursive delete all task instance by process instance id
+     * @param processInstanceId
+     */
+    public void deleteWorkTaskInstanceByProcessInstanceId(int processInstanceId) {
+        List<TaskInstance> taskInstanceList = findValidTaskListByProcessId(processInstanceId);
+        if (CollectionUtils.isEmpty(taskInstanceList)) {
+            return;
+        }
+
+        for (TaskInstance taskInstance : taskInstanceList) {
+            taskInstanceMapper.deleteById(taskInstance.getId());
+            removeTaskLogFile(taskInstance);
+        }
+    }
+
+    /**
+     * remove task log file by task instance
+     *
+     * @param taskInstance taskInstance
+     */
+    public void removeTaskLogFile(TaskInstance taskInstance) {
+        try (LogClientService logClient = new LogClientService()) {
```

Comment:
It seems that it can be operated by batch and don't have to create logClient and release every time.

Reply:
Yes, I'll fix it.